spinalcordtoolbox/reports/qc2.py

```diff
@@ -0,0 +1,352 @@
+                'plane': plane,
+                'background_img': str(path_background_img.relative_to(path_qc)),
+                'overlay_img': str(path_overlay_img.relative_to(path_qc)),
+                'moddate': mod_date.strftime("%Y-%m-%d %H:%M:%S"),
+                'qc': '',
+            }, file_result, indent=1)
+
+        # Collect all existing QC report entries
+        json_data = []
+        for path in sorted(path_json.glob('*.json')):
+            with path.open() as file:
+                json_data.append(json.load(file))
+
+        # Insert the QC report entries into index.html
+        path_assets = importlib.resources.files(spinalcordtoolbox.reports) / 'assets'
+        template = string.Template((path_assets / 'index.html').read_text(encoding='utf-8'))
+        # Empty the HTML file before writing, to make sure there's no leftover junk at the end
+        file_index_html.truncate()
+        file_index_html.write(template.substitute(sct_json_data=json.dumps(json_data)))
+
+        # Copy any missing assets
+        path_qc.mkdir(parents=True, exist_ok=True)
+        copy_missing(path_assets / '_assets', path_qc)
+
+    logger.info('Successfully generated the QC results in %s', str(path_result))
+    display_open(file=str(path_index_html), message="To see the results in a browser")
+
+
+def copy_missing(resource: Traversable, destination: Path):
+    """
+    Make sure that a copy of `resource` exists at `destination`,
+    by creating any missing files and directories recursively.
+    """
+    path = destination / resource.name
+    if resource.is_dir():
+        path.mkdir(exist_ok=True)
+        for sub_resource in resource.iterdir():
+            copy_missing(sub_resource, path)
+    elif resource.is_file():
+        if not path.is_file():
+            path.write_bytes(resource.read_bytes())
+    else:
+        # Some weird kind of resource? Ignore it
+        pass
+
+
+def sct_register_multimodal(
+    fname_input: str,
+    fname_output: str,
+    fname_seg: str,
+    argv: Sequence[str],
+    path_qc: str,
+    dataset: Optional[str],
+    subject: Optional[str],
+):
+    """
+    Generate a QC report for sct_register_multimodal.
+    """
+    command = 'sct_register_multimodal'
+    cmdline = [command]
+    cmdline.extend(argv)
+
+    # Axial orientation, switch between two input images
+    with qc_entry(
+        path_input=Path(fname_input).absolute(),
+        path_qc=Path(path_qc),
+        command=command,
+        cmdline=list2cmdline(cmdline),
+        plane='Axial',
+        dataset=dataset,
+        subject=subject,
+    ) as (path_background_img, path_overlay_img):
+
+        # Resample images slice by slice
+        p_resample = 0.6
+        logger.info('Resample images to %fx%f mm', p_resample, p_resample)
+        img_input = Image(fname_input).change_orientation('SAL')
+        img_input = resample_nib(
+            image=img_input,
+            new_size=[img_input.dim[4], p_resample, p_resample],
```

Comment:
I think this line is one of the main differences between `Axial` and `Sagittal`.

Reply:
Re: This comment + https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4224#discussion_r1333659259:

In other words, once we convert some additional QC methods (both `Axial` and `Sagittal`), we should be able to re-use this code by tweaking the order of `new_size` according to `plane`?

Or, is "trying to reuse the resampling code" part of the reason why the QC code got so hard to work with in the first place? (Since, we enforced a very inflexible structure on all of the various image-generating methods.)

Reply:
Both! I very much think the singular resampling pipeline was a big source of inflexibility, but of course it's tempting to parametrize by `plane`. We should be mindful if/when/how we do it. I can think of a few different ways, for example:
- Having a `resample_slices` function which literally takes a stringly-typed `plane` argument. This might be convenient if we have several places where we only know whether we want axial or sagittal at runtime. But I suspect we'll only need this once, in `sct_qc.py`, in which case we don't gain much by abstracting essentially just an `if` statement.
- Having a pair of functions `resample_axial` and `resample_sagittal` which are very simple wrappers around `resample_nib`. Possibly also combined with the call to `change_orientation`? Something to think about. (Could be informed by the discussion in this [interesting talk on API design](https://caseymuratori.com/blog_0024).)

One thing to note is that only the first image needs this `new_size` parameter! All other images use the first resized image as a template, and don't depend on the axial/sagittal distinction. (For resampling. Later on, when iterating over slices, the axial/sagittal distinction re-appears for all images.)

Reply:
Also, for this refactoring I'm trying to first port some more kinds of reports, and only create functions and function parameters once they're actually used in a few places. That way we'll have some real examples of usage to guide the API design.